src/test/ui/const-generics/projection-as-arg-const.rs

```diff
@@ -0,0 +1,27 @@
+// run-pass
+
+#![crate_name = "foo"]
+#![feature(adt_const_params)] // This is needed to use projections as const generic.
```

Comment:
It seems wrong to me that this is needed, monomorphic projections should be just fine.

Reply:
It's because I only added this support in the branch behind the `adt_const_param` feature check. I can normalize before both though.